src/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -125,5 +114,64 @@ public static IEnumerable<string> GetAllProjectFileNames(string solutionFile)
                 throw new CommandLineException(message);
             }
         }
+
+        public static string GetMsbuildDirectory(string version)
+        {
+            if (String.IsNullOrEmpty(version))
+            {
+                // default to 4.0
+                version = "4.0";
+
+                // run msbuild to get the version
+                var processStartInfo = new ProcessStartInfo
+                {
+                    UseShellExecute = false,
+                    FileName = "msbuild.exe",
+                    Arguments = "/version",
+                    RedirectStandardOutput = true
+                };
+                try
+                {
+                    using (var process = Process.Start(processStartInfo))
+                    {
+                        process.WaitForExit(100);
+                        if (process.ExitCode == 0)
+                        {
+                            var output = process.StandardOutput.ReadToEnd();
+                            var lines = output.Split(
+                                new[] { Environment.NewLine },
+                                StringSplitOptions.RemoveEmptyEntries);
+                            version = lines[lines.Length - 1];
+                        }
+                    }
+                }
+                catch
+                {
+                }
+            }
+
+            var ver = version.Contains('.') ?
+                new Version(version) :
```

Comment:
The output I get is: Microsoft (R) Build Engine version 14.0.23107.0

Is Version able to parse this entire line? It looks like more parsing is needed.

Since this is always very error prone I would move this parsing to another static method and add unit tests just for that, along with tests for garbage input incase the msbuild.exe on the path isn't actually msbuild.